sycl/doc/extensions/proposed/sycl_ext_codeplay_cuda_cluster_group.asciidoc

```diff
@@ -0,0 +1,421 @@
+ensured by synchronizing all the work-items within the cluster before and after
+the local memory operations, using `group_barrier`.
+
+A member function of the `cluster_group` class; 
+`map_cluster_local_pointer` will perform the mapping and return a pointer 
+which can then be dereferenced by the calling work-item. 
+
+
+[%header,cols="10,5"]
+|===
+|Method
+|Description
+
+|T*  map_cluster_local_pointer(T* addr, size_t group_id)
+|Accepts the equivalent address to the memory location relative to the calling 
+work-item which is to be mapped from the local memory of the work-group, as 
+specified by the group_id, denoting the linear group_id within the cluster
+|===
+
+Conversely, `get_linear_cluster_group_id_for_local_pointer` will return the 
+linearized id of the work-group a mapped local memory address belongs to.
+
+[%header,cols="10,5"]
+|===
+|Method
+|Description
+
+|size_t  get_linear_cluster_group_id_for_local_pointer(T* addr)
+|Accepts a mapped pointer to the local memory of another work-group and returns
+the linearized id of the work-group in the cluster from whom's local memory the
+mapped pointer was obtained
+|===
+
+
+== Cluster Memory Fence Scope and Barrier
+
+Work-items in a work-group can access a local memory address from another
+work-group in the cluster-group, which has been mapped as described above. To
+facilitate this, a new memory scope is introduced to the `memory_scope` class;
+`ext_codeplay_cuda_cluster_group` which indicates a memory ordering
+constraint that applies to all work-items in the same cluster-group. This memory
+scope can be used with `atomic_ref`` and other SYCL APIs that use
+`memory_scope`.
+
+[source, c++]
+----
+namespace sycl {
+
+  enum class memory_scope {
+  ...
+  ext_codeplay_cuda_cluster_group,
+  ...
+  };
+
+  namespace ext::codeplay::experimental::cuda {
+    inline constexpr auto memory_scope_cluster_group
+      = memory_scope::ext_codeplay_cuda_cluster_group;
+  } // namespace ext::codeplay::experimental::cuda
+
+----
+
+namespace ext::codeplay::experimental::cuda {
+inline constexpr auto memory_scope_cluster_group = memory_scope::ext_codeplay_cuda_cluster_group;
+} // namespace ext::codeplay::experimental::cuda
+
+
+[source,c++]
+----
+    class cluster_group {
+    public:
+    ...
+    static constexpr memory_scope fence_scope = 
+                memory_scope::ext_codeplay_cuda_cluster_fence;
+    }
+----
+
+To synchronize all workitems in the cluster group, `sycl::group_barrier` can be 
+used, accepting the `cluster_group` class.
+
+
+== Example
+
+This section adds a representative example of how to launch a kernel with 
+the cluster-range specified and accessing various id's within the kernel - 
+
+[source,c++]
+----
+sycl::event launch_kernel_with_cluster() {
+  using cluster_size_prop =
+    sycl::ext::codeplay::experimental::cuda::cluster_size
+  using namespace sycl::ext::codeplay::experimental::cuda;
```

Comment:
Our BKM for examples is to never use `using namespace` because it becomes unclear which identifiers are in which namespace.  Instead, we usually define namespace aliases, and then show fully qualified names.  For example, something like:

```
namespace syclcp = sycl::ext::codeplay::experimental;
namespace syclex = sycl::ext::oneapi::experimental;

sycl::event launch_kernel_with_cluster() {
  sycl::queue q;

  sycl::nd_range<3> ndRange({4096, 4096, 32}, {32, 32, 1});
  syclex::properties props(syclcp::cuda::cluster_size({4, 4, 1}));
  syclex::launch_config config(ndRange, props);

  return syclex::submit_with_event(q, [&](sycl::handler& cgh){
      syclex::nd_launch(cgh, config, [=](sycl::nd_item<3> it) {
        auto cg = it.ext_codeplay_cuda_get_cluster_group();
        auto cgId = cg.get_group_id();
        ...
      });
  })
}
```
```

Reply:
Done

Reply:
I don't see any change to the example.  Maybe you forgot to push a commit?

Reply:
It got stashed when pulling the committed suggested changes, and in my mind, I already had committed it, 
hence it went missing..

This should be fixed now in - [7bc66b2](https://github.com/intel/llvm/pull/13594/commits/7bc66b201b0c97d120683a3147e6da8886be9852)

Thanks